extra_geom/detectors.py

```diff
@@ -2295,3 +2295,283 @@ def position_modules_fast(self, data, *args, **kwargs):
+    
+    The first pixel of the first tile (ASIC 0) corresponds to the first pixel
+    in the data `frame [0,0]`, and the first row of this tile corresponds to
+    half of the row line in the data `frame[0,:ncol//2]`.
+    
+    The four tiles are stacked in 2 rows of 2 columns and numbered along with
+    the columns:
+    [ASIC 0] [ASIC 1]
+    [ASIC 2] [ASIC 3]
+    """
+    # Geometry layout
+    # ---------------
+    # oooo-  -oooo
+    # oooo-  -oooo
+    # ||||#  #||||
+    #
+    # ||||#  #||||
+    # oooo-  -oooo
+    # oooo-  -oooo
+    #
+    # o : normal pixels 50x50 um
+    # - : long horizontal pixels 50x175 um
+    # | : long vertical pixels 175x50 um
+    # # : big square pixels 175x175 um (4 big pixels in the center)
+    #
+    # pixel sizes given for ePix100, ePix10K has doubled size pixels
+    #
+    # Electronic layout
+    # -----------------
+    # detector: 704+4 rows, 768 columns, 4 asics
+    # asic: 352+2 rows, 384 columns, 4 banks
+    # bank: 352+2 rows, 96 columns
+    #     - fast parallel column readout
+    #     - sigma-delta ADC per column
+    #     - single high speed LVDS link
+    #
+    # (+2) Each asics has two calibration rows:
+    #     - pixel max (the first and last rows in the pixel array)
+    #     - baseline (next rows)
+    #
+    # Schema
+    # ------
+    #      ||||||  ||||||
+    #   =                 =
+    #   =    A2      A1   =  352+2
+    #
+    #   =    A3      A0   =  352+2
+    #   =                 =
+    #      ||||||  ||||||
+    #        384     384
+    #
+    # Conversions in electronic layout
+    # -------------------------------
+    # - ASIC position to No.:
+    #     tileno = 2*(1 - row*col) + row - col
+    #
+    # - ASIC No. to position:
+    #     hi, lo = tileno // 2, tileno % 2
+    #     i, j = (hi - lo + 1) & 1, 1 - hi
+    #
+    # - ASIC slice:
+    #     [(nrow-1)*row : nrow//2-row : 1-2*row] or [-1 : nrow//2-row : 1-2*row]
+    #     [(ncol-1)*col : ncol//2-col : 1-2*col] or [-1 : ncol//2-col : 1-2*col]
+    #
+    # See also
+    # --------
+    # 1. [EuXFEL ePix documentation]
+    #     (https://in.xfel.eu/readthedocs/docs/epix-documentation/en/latest/index.html)
+    # 2. [A Dragone et al 2014 J. Phys.: Conf. Ser. 493 012012]
+    #     (https://doi.org/10.1088/1742-6596/493/1/012012)
+    # 3. [SLAC Confluence](https://confluence.slac.stanford.edu/display/PSDM/EPIX10KA)
+    #     Describes ePix10KA instead of ePix100, many details are common for these
+    #     two versions
+    # See also
+    # --------
+    # 1. [EuXFEL ePix documentation]
+    #     (https://in.xfel.eu/readthedocs/docs/epix-documentation/en/latest/index.html)
+    # 2. [A Dragone et al 2014 J. Phys.: Conf. Ser. 493 012012]
+    #     (https://doi.org/10.1088/1742-6596/493/1/012012)
+    # 3. [SLAC Confluence](https://confluence.slac.stanford.edu/display/PSDM/EPIX10KA)
+    #     Describes ePix10KA instead of ePix100, many details are common for these
+    #     two versions
```

Comment:
```suggestion
    # See also
    # --------
    # 1. [EuXFEL ePix documentation]
    #     (https://in.xfel.eu/readthedocs/docs/epix-documentation/en/latest/index.html)
    # 2. [A Dragone et al 2014 J. Phys.: Conf. Ser. 493 012012]
    #     (https://doi.org/10.1088/1742-6596/493/1/012012)
    # 3. [SLAC Confluence](https://confluence.slac.stanford.edu/display/PSDM/EPIX10KA)
    #     Describes ePix10KA instead of ePix100, many details are common for these
    #     two versions
```